### PR TITLE
[Trivia] V3 Rewrite

### PR DESCRIPTION
**Pull request by Tobotimus**
This PR supersedes #789 and as such many of the following feature suggestions came from there.

### Goals
- [x] Get cog working in V3
- [x] Rewrite `TriviaSession`
- [x] Consolidate how the trivia lists are stored and read (#917)
- [x] Add a payout option for winning
- [x] Use YAML for trivia lists

### Possible enhancements
- [x] Multiple categories per session
- [x] Implement some sort of leaderboard system
- [ ] Use of embeds
- [ ] MCQ trivia

### CODE REVIEW

**Comment by Tobotimus** [comment]
Alright, `TriviaSession` has just undergone some major changes, and I'm interested to see what you guys think. The back-end of this cog is now fairly complete, so please... try to break it :smile:

_Edit: I probably should have mentioned that although most of the back-end is complete, we are still waiting to see what we are doing with trivia lists._

**Comment by Twentysix26** [comment]
I think you can safely remove the trivia lists from your PR in the meantime. It's unlikely they'll be kept in this branch and they can eventually be added with a different pull request if needed

**Comment by Tobotimus** [comment]
That's fair enough, they were only there for testing purposes, so no real reason for them to be in the repository, considering they would only be there temporarily.

But on that note, if anyone is testing this PR, trivia lists must be located in the `cogs/trivia/lists` folder in order to be found (until further notice).

File: redbot/cogs/trivia/session.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+            :code:`True` if the session wasn't interrupted.
+
+        """
+        try:
+            message = await self.ctx.bot.wait_for(
+                "message", check=self.check_answer(answers), timeout=delay)
+        except asyncio.TimeoutError:
+            if abs(self._last_response - int(time.perf_counter())) >= timeout:
+                await self.ctx.send("Guys...? Well, I guess I'll stop then.")
+                await self.stop()
```
Comment (calebj) [inline redbot/cogs/trivia/session.py]: `self.stop()` isn't an async function.

File: redbot/cogs/trivia/session.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+            How long before the session ends due to no responses (in seconds).
+
+        Returns
+        -------
+        bool
+            :code:`True` if the session wasn't interrupted.
+
+        """
+        try:
+            message = await self.ctx.bot.wait_for(
```
Comment (calebj) [inline redbot/cogs/trivia/session.py:159]: I don't see anything here that would cause the issue you described. Try to debug and isolate it.

**Review by calebj — CHANGES_REQUESTED** [review]

**Comment by Tobotimus** [comment]
I think this PR is ready for proper testing. Any extra features should probably be added in separate PRs.

To test right now, I recommend doing the following in a clean virtual environment:
```
pip install --process-dependency-links git+https://github.com/Tobotimus/Red-DiscordBot.git@V3/trivia_rewrite
```
Then run redbot as usual with `redbot <instance_name>`

**Review by tekulvw — CHANGES_REQUESTED** [review]
Can you add `Red-Trivia` to the requirements for the bot? I'll get it uploaded to pypi soon.

**Review by tekulvw — DISMISSED** [review]

**Review by palmtree5 — APPROVED** [review]

**Pull request merged**